File: Ruby/parse.rb
module Parse
  def self.data(year, day, int=false)
    data = File.open("#{year}/#{day}/data.txt", "r")
    lines = data.readlines.map do |line|
      line.chomp.split(" ")
    end
    data.close
    int ? lines.map { |line| line.map(&:to_i) } :
    lines
  end

  def self.test(year, day, int=false)
    data = File.open("#{year}/#{day}/test.txt", "r")
    lines = data.readlines.map do |line|
      line.chomp.split(" ")
    end
    data.close
    int ? lines.map { |line| line.map(&:to_i) } :
    lines
  end
end

File: Python/parse.py
def parse(
        fileName: str,
        integer: bool = False,
        liste: bool = False,
        texte: bool = False
    ):
    # lecture du fichier
    with open(fileName, "r") as f:
        if texte:
            with open(fileName, "r") as f:
                return f.read()
        else:
            L = []
            # éclater chaque line en liste
            for line in [line.split() for line in f.readlines() if line.split() != []]:
                # si ce sont des entiers
                if integer:
                    L.append([int(x) for x in line]) # changer en entier
                # sinon ne rien faire
                else:
                    L.append(line)

            # si une liste est attendue, chaque ligne n'a qu'un seu élément
            if liste:
                L = [line[0] for line in L]
        return L
